CMakeLists.txt

```diff
@@ -0,0 +1,250 @@
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
+    CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 5.0)
+    option(OPENASSETIO_GLIBCXX_USE_CXX11_ABI "For gcc, use the new C++11 library ABI" OFF)
+endif ()
+
+# Whether to use RPATH or RUNPATH runtime search path behavior.
+if (IS_GCC_OR_CLANG)
+    option(OPENASSETIO_ENABLE_NEW_DTAGS "If enabled, RUNPATH will be set, overriding RPATH" OFF)
+endif ()
+
+# Testing instrumentation.
+if (IS_GCC_OR_CLANG)
+    # Coverage reporting (for tests).
+    option(OPENASSETIO_ENABLE_COVERAGE "Enable coverage reporting for gcc/clang" OFF)
+
+    # Sanitizers (primarily for tests)
+    option(OPENASSETIO_ENABLE_SANITIZER_ADDRESS "Enable address sanitizer" OFF)
+    option(OPENASSETIO_ENABLE_SANITIZER_LEAK "Enable leak sanitizer" OFF)
+    option(OPENASSETIO_ENABLE_SANITIZER_UNDEFINED_BEHAVIOR
+        "Enable undefined behavior sanitizer" OFF)
+    option(OPENASSETIO_ENABLE_SANITIZER_THREAD "Enable thread sanitizer" OFF)
+    option(OPENASSETIO_ENABLE_SANITIZER_MEMORY "Enable memory sanitizer" OFF)
+endif ()
+
+
+#-----------------------------------------------------------------------
+# Enable ctest testing
+
+# To correctly generate the DartConfiguration.tcl, include the CTest
+# CMake file. By doing so 'ctest -F memcheck' works.
+# TODO(DF): ...but also creates many many pointless targets.
+#include(CTest)  # Automatically calls enable_testing()
+enable_testing()
+
+#-----------------------------------------------------------------------
+# Provides install directory variables as defined by the GNU Coding
+# Standards.
+
+include(GNUInstallDirs)
+
+#-----------------------------------------------------------------------
+# Generate OpenAssetIOConfig.cmake et al. for bundling as a CMake lib.
+
+# Targets exported to ${PROJECT_NAME}_EXPORTED_TARGETS will be available
+# for import using find_package(OpenAssetIO).
+
+include(CMakePackageConfigHelpers)
+
+set(version_config_file "${PROJECT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake")
+set(project_config_file "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake")
+set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+
+# Version fetched from the top level project()
+write_basic_package_version_file(
+    ${version_config_file}
+    # Version range supported only if generated by CMake >= 3.19.2.
+    COMPATIBILITY SameMinorVersion
+)
+
+configure_package_config_file(
+    cmake/Config.cmake.in
+    ${project_config_file}
+    INSTALL_DESTINATION ${config_install_dir}
+    # TODO(DF): PATH_VARS?
+)
+
+install(
+    EXPORT ${PROJECT_NAME}_EXPORTED_TARGETS
+    DESTINATION ${config_install_dir}
+    NAMESPACE ${PROJECT_NAME}::
+    FILE ${PROJECT_NAME}Targets.cmake
+)
+
+install(
+    FILES "${project_config_file}" "${version_config_file}"
+    DESTINATION "${config_install_dir}"
+)
+
+# TODO(DF): pkg-config? See OCIO.
+
+
+#-----------------------------------------------------------------------
+# Print a status dump
```

Comment:
`OPENASSETIO_ENABLE_NEW_DTAGS` is there.  

I wasn't going to put options that are only likely used internally for testing.  However, You make a good point in https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/185#discussion_r786824884 that it would be worth reporting if coverage (and/or sanitizers, I guess) are enabled.  Will add those.

Reply:
Added additional `STATUS` messages in 9ef4f8bf and added a `NOTICE` message specifically for coverage in fe213317